File: JAVA/二年前期/Lesson07_Mon01/Itstudent.java
// Itstudent クラスは StudentCard クラスを継承（サブクラス）
class Itstudent extends StudentCard {
    String gakka;        // 学科名
    String nationality;  // 国籍

    // コンストラクタ：学籍番号・氏名・学科名・国籍を初期化
    Itstudent(int num, String name, String gakka, String nationality) {
        super(num, name);             // 親クラスのコンストラクタを呼び出す
        this.gakka = gakka;
        this.nationality = nationality;
    }

    // 学生情報を表示するメソッド（オーバーライド）
    void printInfo() {
        System.out.println();
        System.out.println("ーーーー　学生情報　－－－－");
        System.out.println("学 科 名：" + this.gakka);   // 学科名を表示
        super.printInfo();                                // 親クラスの情報（学籍番号・氏名）を表示
        System.out.println("　国　籍：" + this.nationality); // 国籍を表示
    }
}

File: Python/二年前期/Assignments(練習問題）/0627/Itstudent.py
from superclass import StudentCard  # 親クラスStudentCardをインポート

# StudentCardクラスを継承したItstudentクラス
class Itstudent(StudentCard):
    # コンストラクタ（学籍番号・学科名・出席回数を初期化）
    def __init__(self, number, gakka, attendees):
        super().__init__(number, gakka)  # 親クラスの初期化
        self.attendees = attendees       # 出席回数
        self.rate = 0                    # 出席率（初期値は0）

    # 出席率を計算するメソッド（全15回中の割合）
    def get_rate(self):
        self.rate = int(self.attendees / 15 * 100)

    # 学籍情報と出席率を表示するメソッド
    def show_info(self):
        print()
        super().show_info()              # 親クラスのshow_info()を呼び出す
        print("出席率:", self.rate, "%")
